packages/react-dom/src/__tests__/ReactDOMConsoleErrorReporting-test.js

```diff
@@ -241,32 +249,44 @@ describe('ReactDOMConsoleErrorReporting', () => {
       }
 
       const root = ReactDOMClient.createRoot(container);
-      await act(async () => {
+      await fakeAct(() => {
         root.render(<Foo />);
-        await waitForThrow('Boom');
       });
 
       if (__DEV__) {
-        expect(windowOnError.mock.calls).toEqual([]);
+        expect(windowOnError.mock.calls).toEqual([
+          [
+            expect.objectContaining({
+              message: 'Boom',
+            }),
+          ],
+        ]);
         expect(console.error.mock.calls).toEqual([
           [
-            // Formatting
-            expect.stringContaining('%o'),
             expect.objectContaining({
               message: 'Boom',
             }),
+          ],
+        ]);
+        expect(console.warn.mock.calls).toEqual([
```

Comment:
We're moving some logs from `console.error` to `console.warn` right? Can we add that to the "Breaking changes" in the description?

Reply:
So it used to be that we had two logs. One for the error and then one for the suffix. In my previous PR removing invokeGuardedCallback I was able to unify them into one but what this is doing is basically restoring it into two because we can't pass a component stack to reportGlobalError for uncaught but we can for caught which just goes to console.error.

It really isn't great to have the two logs, especially if the former gets called with preventDefault which used to silence the second one but we can't do that anymore.

I'm kind of inclined to say that we shouldn't have this second log at all and let uncaught not have component stacks by default and instead if you want that you implement it yourself using `onUncaughtError`.

Reply:
I like it. But why switch from `error` to `warn`?

Would using `onUncaughtError` result in double logging or would we not use reportGlobalError if that's provided?

Reply:
The `warn` is mainly about telling you to add an error boundary. It's not really so much about the component stack anymore although it includes it. Where as before it was more like two `console.error` as a pair. The `console.warn` can stand on its own.

You should really always have a root error boundary and if you don't want to you should at least implement onUncaughtError. So this shouldn't really be a case you need to hit beyond seeing the warning.

onUncaughtError takes control so this doesn't get logged then.